src/OpenSage.Mods.Generals/Gui/GeneralsControlBar.cs

```diff
@@ -313,8 +325,48 @@ public override void Update(Player player, GeneralsControlBar controlBar)
 
                 var unitSelectedControl = controlBar._right.Controls.FindControl("ControlBar.wnd:WinUnitSelected");
 
+                var productionQueueWindow = controlBar._right.Controls.FindControl("ControlBar.wnd:ProductionQueueWindow");
+                productionQueueWindow.Visible = unit.IsProducing;
+
+                var queue = unit.ProductionQueue;
+
+                for (var pos = 0; pos < PRODUCTION_QUEUE_SIZE; pos++)
+                {
+                    var queueButton = productionQueueWindow.Controls.FindControl($"ControlBar.wnd:ButtonQueue0{pos+1}");
+
+                    if(queueButton == null)
+                    {
+                        logger.Warn($"Could not find the right control (ControlBar.wnd:ButtonQueue0{pos+1})");
+                        continue;
+                    }
+
+                    Image img = null;
+                    if (queue.Count > pos)
+                    {
+                        var job = queue[pos];
+                        if (queueButton != null && job != null)
```

Comment:
Since you check `queueButton` for null earlier, you can remove the check from here.

```suggestion
                        if (job != null)
```